src/autopas/selectors/AutoTuner.h

```diff
@@ -343,59 +391,53 @@ bool AutoTuner<Particle, ParticleCell>::tune(PairwiseFunctor &pairwiseFunctor) {
   // repeat as long as traversals are not applicable or we run out of configs
-  do {
-    _containerSelector.selectContainer(_currentConfig->_container);
-
-    // check if newton3 works with this functor and fix if needed
+  while (_currentConfig != _allowedConfigurations.end()) {
+    // check if newton3 works with this functor and remove config if not
     if ((_currentConfig->_newton3 == Newton3Option::enabled and not pairwiseFunctor.allowsNewton3()) or
         (_currentConfig->_newton3 == Newton3Option::disabled and not pairwiseFunctor.allowsNonNewton3())) {
       AutoPasLog(warn, "Configuration with newton 3 {} called with a functor that does not support this!",
                  utils::StringUtils::to_string(_currentConfig->_newton3));
 
-      Configuration modifiedCurrentConfig = *_currentConfig;
-      // choose the other option
-      modifiedCurrentConfig._newton3 = Newton3Option((static_cast<int>(_currentConfig->_newton3) + 1) % 2);
-
-      // if modified config is equal to next delete current. Else insert modified config.
-      // the disabled case should come after the enabled.
-      int searchDirection = _currentConfig->_newton3 == Newton3Option::enabled ? 1 : -1;
-      if (modifiedCurrentConfig == *(std::next(_currentConfig, searchDirection))) {
-        AutoPasLog(warn, "Newton 3 {} automatically for this configuration!",
-                   utils::StringUtils::to_string(modifiedCurrentConfig._newton3));
-        if (pairwiseFunctor.isRelevantForTuning()) {
-          _currentConfig = _allowedConfigurations.erase(_currentConfig);
-        }
+      _currentConfig = _allowedConfigurations.erase(_currentConfig);
+    } else {
+      if (configApplicable(*_currentConfig, pairwiseFunctor)) {
+        // we found a valid config!
+        break;
       } else {
-        _currentConfig = _allowedConfigurations.insert(_currentConfig, modifiedCurrentConfig);
-      }
-      if (_currentConfig != _allowedConfigurations.end()) {
-        _containerSelector.selectContainer(_currentConfig->_container);
+        ++_currentConfig;
       }
     }
-
-    // if current config is not applicable but there are still some left check next
-  } while (_currentConfig != _allowedConfigurations.end() and not configApplicable(*_currentConfig, pairwiseFunctor) and
-           ((++_currentConfig) != _allowedConfigurations.end()));
+  }
 
   // reached end of tuning phase
   // either wait until last config has enough samples or last config is not applicable
-  if (_currentConfig == _allowedConfigurations.end() and (_numSamples >= _maxSamples or not configIsApplicable)) {
-    selectOptimalConfiguration();
-    stillTuning = false;
-    _containerSelector.selectContainer(_currentConfig->_container);
+  if (_currentConfig == _allowedConfigurations.end()) {
+    if (_traversalTimes.empty()) {
+      autopas::utils::ExceptionHandler::exception("AutoTuner: No applicable configurations found!");
+    }
+
+    if (_numSamples >= _maxSamples) {
```

Comment:
should not be needed, there are no samples for the invalid config `_allowedConfigurations.end()`